CRM/Core/Form.php

```diff
@@ -1677,6 +1677,10 @@ public function addField($name, $props = [], $required = FALSE, $legacyDate = TR
         $props['size'] = isset($props['size']) ? $props['size'] : 60;
         return $this->add('password', $name, $label, $props, $required);
 
+      case 'YesNo':
```

Comment:
@seamuslee001 we should work through this part first - the YesNo is a new thing for addField - it seems to be a specialised Radio

Reply:
@eileenmcnaughton yeh its kind of a nicer looking radio i guess or checkbox. 

seems like its used in a number of places

```
CRM/Contact/Form/Search/Criteria.php:    $form->addYesNo('uf_user', ts('CMS User?'), TRUE);
CRM/Contact/Form/Search/Criteria.php:    $form->addYesNo('is_deceased', ts('Deceased'), TRUE);
CRM/Pledge/BAO/Query.php:    $form->addYesNo('pledge_test', ts('Pledge is a Test?'), TRUE);
CRM/Pledge/BAO/Query.php:    $form->addYesNo('pledge_acknowledge_date_is_not_null', ts('Acknowledgement sent?'), TRUE);
CRM/Contribute/BAO/Query.php:    $form->addYesNo('contribution_thankyou_date_is_not_null', ts('Thank-you sent?'), TRUE);
CRM/Contribute/BAO/Query.php:    $form->addYesNo('contribution_receipt_date_is_not_null', ts('Receipt sent?'), TRUE);
CRM/Contribute/BAO/Query.php:    $form->addYesNo('contribution_pay_later', ts('Contribution is Pay Later?'), TRUE);
CRM/Contribute/BAO/Query.php:    $form->addYesNo('contribution_recurring', ts('Contribution is Recurring?'), TRUE);
CRM/Contribute/BAO/Query.php:    $form->addYesNo('contribution_test', ts('Contribution is a Test?'), TRUE);
CRM/Contribute/BAO/Query.php:    $form->addYesNo('contribution_pcp_display_in_roll', ts('Personal Campaign Page Honor Roll?'), TRUE);
CRM/Activity/BAO/Query.php:    $form->addYesNo('activity_test', ts('Activity is a Test?'));
CRM/Member/BAO/Query.php:    $form->addYesNo('membership_is_current_member', ts('Current Member?'), TRUE);
CRM/Member/BAO/Query.php:    $form->addYesNo('member_is_primary', ts('Primary Member?'), TRUE);
CRM/Member/BAO/Query.php:    $form->addYesNo('member_pay_later', ts('Pay Later?'), TRUE);
CRM/Member/BAO/Query.php:    $form->addYesNo('member_test', ts('Membership is a Test?'), TRUE);
CRM/Member/BAO/Query.php:    $form->addYesNo('member_is_override', ts('Membership Status Is Overriden?'), TRUE);
CRM/Event/Form/ManageEvent/Registration.php:      $form->addYesNo('is_confirm_enabled', ts('Use a confirmation screen?'), NULL, NULL, ['onclick' => "return showHideByValue('is_confirm_enabled','','confirm_screen_settings','block','radio',false);"]);
CRM/Event/Form/ManageEvent/Registration.php:    $form->addYesNo('is_email_confirm', ts('Send Confirmation Email?'), NULL, NULL, ['onclick' => "return showHideByValue('is_email_confirm','','confirmEmail','block','radio',false);"]);
CRM/Event/Form/ManageEvent/Fee.php:    $this->addYesNo('is_monetary',
CRM/Event/BAO/Query.php:    $form->addYesNo('participant_test', ts('Participant is a Test?'), TRUE);
CRM/Event/BAO/Query.php:    $form->addYesNo('participant_is_pay_later', ts('Participant is Pay Later?'), TRUE);
CRM/Admin/Form/SettingTrait.php:        elseif ($add === 'addYesNo' && ($props['type'] === 'Boolean')) {
CRM/Admin/Form/Preferences.php:            case 'YesNo':
CRM/Admin/Form/Preferences.php:          case 'YesNo':
CRM/Admin/Form/Setting/Smtp.php:    $this->addYesNo('smtpAuth', ts('Authentication?'), CRM_Utils_Array::value('smtpAuth', $props));
CRM/Admin/Form/Setting/Url.php:    $this->addYesNo('enableSSL', ts('Force Secure URLs (SSL)'));
CRM/Admin/Form/Setting/Url.php:    $this->addYesNo('verifySSL', ts('Verify SSL Certs'));
CRM/Core/Form.php:  public function addYesNo($id, $title, $allowClear = FALSE, $required = NULL, $attributes = []) {
CRM/Case/Form/Activity/ChangeCaseType.php:    $form->addYesNo('is_reset_timeline', ts('Reset Case Timeline?'), NULL, TRUE);
CRM/Mailing/Form/Component.php:    $this->addYesNo('is_default', ts('Default?'));
CRM/Mailing/Form/Component.php:    $this->addYesNo('is_active', ts('Enabled?'));
CRM/Mailing/Form/Search.php:    $this->addYesNo('is_archived', ts('Mailing is Archived'), TRUE);
```

Reply:
@eileenmcnaughton @seamuslee001 I think adding this option for addField extend the ability for boolean fields of the respective entity. Should I move this part in a separate PR ?